mava/systems/ippo/networks.py

```diff
@@ -322,6 +323,7 @@ def critic_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
         """
         critic_network = hk.Sequential(
             [
+                observation_network,
```

Comment:
Are we now assuming a shared observation network if it is provided? However, our trainers are set up for separate networks. Would this not cause issues when two optimiser optimise the same parameters?

Reply:
We use seperate observation networks since Haiku enforces passing in a lambda function for the obs network.

Reply:
Thanks for the update. Separate networks sound better for now, yes.